src/widgets/AutoCompletion/ui/AutoCompletionScreen.tsx

```diff
@@ -0,0 +1,124 @@
+import { useCallback, useContext } from 'react';
+
+import { ProgressBar } from './ProgressBar';
+
+import { appletModel } from '~/entities/applet';
+import { useBanners } from '~/entities/banner/model';
+import { AutoCompletionModel } from '~/features/AutoCompletion';
+import { SurveyContext, SurveyLayout, SurveyManageButtons } from '~/features/PassSurvey';
+import { ROUTES, Theme } from '~/shared/constants';
+import Box from '~/shared/ui/Box';
+import Text from '~/shared/ui/Text';
+import { useCustomNavigation, useCustomTranslation, useOnceEffect } from '~/shared/utils';
+
+export const AutoCompletionScreen = () => {
+  const { t } = useCustomTranslation();
+
+  const navigator = useCustomNavigation();
+
+  const context = useContext(SurveyContext);
+
+  const { addWarningBanner } = useBanners();
+
+  const { entityCompleted } = appletModel.hooks.useGroupProgressStateManager();
+
+  const { removeAutoCompletion } = AutoCompletionModel.useAutoCompletionStateManager();
+
+  const { state, startEntityCompletion, activityName } = AutoCompletionModel.useAutoCompletion();
+
+  const onFinish = useCallback(() => {
+    if (!state) {
+      throw new Error('[AutoCompletionScreen:onFinish] State is not defined');
+    }
+
+    const canBeClosed =
+      state.activityIdsToSubmit.length === state.successfullySubmittedActivityIds.length;
+
+    if (!canBeClosed) {
+      return addWarningBanner(t('answerProcessingScreen.processInProgress'));
+    }
+
+    removeAutoCompletion({
+      entityId: context.entityId,
+      eventId: context.eventId,
+    });
```

Comment:
What happens if the use closes the browser tab instead of clicking the Finish button?

Reply:
Nothing bad. If a user has closed a tab the autoCompletionRecord still will be in the store. The certain group will still be in progress until the user clicks on it and goes to the autoCompletion screen. Then the user should click the "close button".
I have provided the video. You can check how it will look